src/Method.php

```diff
@@ -29,27 +29,30 @@
 
     public static function fromName(string $name): Method
     {
-        $method = Method::tryFromName($name);
-        if (is_null($method)) {
-            $enumName = static::class;
-            throw new ValueError("$name is not a valid name for enum \"$enumName\"");
+        $method = self::tryFromName($name);
+
+        if (null === $method) {
+            throw new ValueError(sprintf('%s is not a valid name for enum "%s"', $name, static::class));
         }
+
         return $method;
     }
     public static function tryFromName(?string $name): ?Method
     {
-        return match ($name) {
-            'GET' => Method::GET,
-            'HEAD' => Method::HEAD,
-            'POST' => Method::POST,
-            'PUT' => Method::PUT,
-            'DELETE' => Method::DELETE,
-            'CONNECT' => Method::CONNECT,
-            'OPTIONS' => Method::OPTIONS,
-            'TRACE' => Method::TRACE,
-            'PATCH' => Method::PATCH,
-            null => null,
-            default => null,
+        if (null === $name) {
+            return null;
+        }
+
+        return match (strtoupper($name)) {
+            'GET' => self::GET,
+            'HEAD' => self::HEAD,
+            'POST' => self::POST,
+            'PUT' => self::PUT,
+            'DELETE' => self::DELETE,
+            'CONNECT' => self::CONNECT,
+            'OPTIONS' => self::OPTIONS,
+            'TRACE' => self::TRACE,
+            'PATCH' => self::PATCH,
```

Comment:
You removed the default case, which will change all previously returned null values to UnhandledMatchError Exceptions. I'm not sure you made that on purpose.

Reply:
No, I moved it out of the `match` because I added a `strtoupper` in the match expression.

Reply:
Hmm, but that's a public method and someone could pass it something like 'NOTAMETHOD' which would change the behaviour. Instead of returning null it would throw an exception.

Reply:
There is a test in the testsuite which specifically tests for this situation which is now failing.

The `try` methods should return a `null` value when an invalid value has been given and not throw an error, (unless it' s a type error) while the equivalent method without the `try` prefix should be throwing an exception upon encountering an invalid value.

Reply:
True, I just pushed a fix.